packages/cactus-common/src/test/typescript/unit/js-object-signer.test.ts

```diff
@@ -0,0 +1,210 @@
+
+  const date: Date = new Date();
+
+  const inner1 = {
+    someProperty: "cool",
+    otherStuff: "also cool",
+    dateProperty: date,
+  };
+  const outer1 = {
+    innerProperty: inner1,
+    outerProperty: "test",
+    outerDateProperty: date,
+  };
+  const sign1 = jsObjectSigner.sign(outer1);
+
+  const inner2 = {
+    dateProperty: date,
+    otherStuff: "also cool",
+    someProperty: "cool",
+  };
+  const outer2 = {
+    outerDateProperty: date,
+    outerProperty: "test",
+    innerProperty: inner2,
+  };
+  const sign2 = jsObjectSigner.sign(outer2);
+
+  assert.equals(sign1.toString, sign2.toString);
+});
+
+test("Circular JSON Test", async (assert: any) => {
+  const jsObjectSignerOptions: IJsObjectSignerOptions = {
+    privateKey: keyPairs.privateKey,
+    logLevel: "debug",
+  };
+  const jsObjectSigner = new JsObjectSigner(jsObjectSignerOptions);
+
+  const date: Date = new Date();
+
+  const obj: any = { a: "foo" };
+  obj.b = obj;
+
+  assert.throws(() => jsObjectSigner.sign(obj));
+});
+
+test("Very Signature Test", async (assert: any) => {
+  const jsObjectSignerOptions: IJsObjectSignerOptions = {
+    privateKey: keyPairs.privateKey,
+    logLevel: "debug",
+  };
+  const jsObjectSigner = new JsObjectSigner(jsObjectSignerOptions);
+
+  const payload1 = { field1: "test11", field2: "test12", field3: 13 };
+  const sign1 = jsObjectSigner.sign(payload1);
+
+  const verify = jsObjectSigner.verify(payload1, sign1, keyPairs.publicKey);
+
+  assert.equals(true, verify);
+});
+
+test("Test optional sign function", async (assert: any) => {
+  const jsObjectSignerOptions: IJsObjectSignerOptions = {
+    privateKey: keyPairs.privateKey,
+    logLevel: "debug",
+    signatureFunc: signFunction,
+  };
+
+  const jsObjectSigner = new JsObjectSigner(jsObjectSignerOptions);
+
+  const inner1 = { someProperty: "cool", otherStuff: "also cool" };
+  const outer1 = { innerProperty: inner1, outerProperty: "test" };
+  const sign1 = jsObjectSigner.sign(outer1);
+
+  const inner2 = { otherStuff: "also cool", someProperty: "cool" };
+  const outer2 = { outerProperty: "test", innerProperty: inner2 };
+  const sign2 = jsObjectSigner.sign(outer2);
+
+  assert.equals(sign1.toString, sign2.toString);
+});
+
+test("Test optional verify sign function", async (assert: any) => {
+  const jsObjectSignerOptions: IJsObjectSignerOptions = {
+    privateKey: keyPairs.privateKey,
+    logLevel: "debug",
+    signatureFunc: signFunction,
+    verifySignatureFunc: verifySignFunction,
+  };
+
+  const jsObjectSigner = new JsObjectSigner(jsObjectSignerOptions);
+
+  const inner1 = { someProperty: "cool", otherStuff: "also cool" };
+  const outer1 = { innerProperty: inner1, outerProperty: "test" };
+  const sign1 = jsObjectSigner.sign(outer1);
+
+  const verify = jsObjectSigner.verify(outer1, sign1, keyPairs.publicKey);
+
+  assert.equals(true, verify);
+});
+
+test("Test optional hash function", async (assert: any) => {
```

Comment:
The `assert` parameter can be typed by importing the `Test` type from the `tape` library using the brackets syntax like this:
```typescript
import test, { Test } from "tape";
```

If you do this then the IDE will have auto complete for all the possible assertion methods that the testing library provides (big help when you don't have to look up stuff like how do they call the deepEquals assertion method and other things like that.

Reply:
Update parameter type.